plugins/web/opentelemetry-plugin-document-load/src/meterDocumentLoad.ts

```diff
@@ -0,0 +1,101 @@
+import { BaseMetricPlugin } from '@opentelemetry/core/build/src/platform/browser';
+import { otperformance } from '@opentelemetry/core';
+import { MetricPluginConfig } from '@opentelemetry/api';
+import { MetricObservable } from '@opentelemetry/metrics';
+import {
+    hasKey,
+    PerformanceEntries,
+    PerformanceLegacy,
+    PerformanceTimingNames as PTN,
+  } from '@opentelemetry/web';
+
+
+export class meterDocumentLoad extends BaseMetricPlugin<unknown> {
+    constructor(config: MetricPluginConfig = {}) {
+        super('@opentelemetry/plugin-document-load');
+        this._onDocumentLoaded = this._onDocumentLoaded.bind(this);
+        this._config = config;
+    }
+
+    private _onDocumentLoaded() {
+        // Timeout is needed as load event doesn't have yet the performance metrics for loadEnd.
+        // Support for event "loadend" is very limited and cannot be used
+        window.setTimeout(() => {
+            this._collectPerformance();
+        });
+    }
+
+    private _collectPerformance() {
+        const entries = this._getEntries();
+        console.log(entries);
+        const loadTimeObserver = this._meter.createObserver('document-load', {
+            description: 'Documet load time',
+        });
+        const loadTime = new MetricObservable();
+        loadTimeObserver.setCallback((observerResult) => {
+            observerResult.observe(loadTime, {});
+        });
+        loadTime.next(Number(entries[PTN.LOAD_EVENT_END]) - Number(entries[PTN.FETCH_START]));
+    }
+
+    private _getEntries() {
```

Comment:
```suggestion
    private _getEntries(): PerformanceEntries {
```

Reply:
I prefer to utilize return types so I can tell at a glance what my expectation should be when using the function.